server/src/main/java/org/jboss/as/server/moduleservice/ModuleDefinition.java

```diff
@@ -25,16 +25,31 @@ public class ModuleDefinition {
     private final Set<ModuleDependency> dependencies;
     private final ModuleSpec moduleSpec;
 
+
+    /** @deprecated use {@link #ModuleDefinition(String, Set, ModuleSpec)} */
+    @Deprecated(forRemoval = true)
     public ModuleDefinition(final ModuleIdentifier moduleIdentifier, final Set<ModuleDependency> dependencies, final ModuleSpec moduleSpec) {
         this.moduleIdentifier = moduleIdentifier;
         this.dependencies = dependencies;
         this.moduleSpec = moduleSpec;
     }
 
+    public ModuleDefinition(final String moduleIdentifier, final Set<ModuleDependency> dependencies, final ModuleSpec moduleSpec) {
+        this.moduleIdentifier = ModuleIdentifier.fromString(moduleIdentifier);  // inefficient but this is unused. When we switch the use to this we'll store the string.
+        this.dependencies = dependencies;
+        this.moduleSpec = moduleSpec;
+    }
+
+    /** @deprecated use {@link #getModuleName()}  */
+    @Deprecated(forRemoval = true)
```

Comment:
As a rule of thumb, we are always asking for the since attribute on all @Deprecated annotations so we can make an idea without looking at the Git history when it was deprecated.

This can be done later though